apps/token/src/routes/governance/propose/update-market/propose-update-market.spec.tsx

```diff
@@ -0,0 +1,182 @@
+      },
+    ],
+    loading: false,
+    error: undefined,
+  }),
+}));
+
+jest.mock('@apollo/client', () => ({
+  ...jest.requireActual('@apollo/client'),
+  useQuery: jest.fn(() => ({
+    data: {
+      marketsConnection: {
+        __typename: 'MarketConnection',
+        edges: [
+          {
+            __typename: 'MarketEdge',
+            node: {
+              __typename: 'Market',
+              id: 'd2319c91104a523032cf04ac4e20962b87ee1f187d1e411a2cac12554dd38f29',
+              tradableInstrument: {
+                __typename: 'TradableInstrument',
+                instrument: {
+                  __typename: 'Instrument',
+                  name: 'ETHUSD (September Market)',
+                  code: 'ETHiUSDT',
+                },
+              },
+            },
+          },
+          {
+            __typename: 'MarketEdge',
+            node: {
+              __typename: 'Market',
+              id: 'bb941f84e25b3a06068e33917d01215d56a51e9abff6ff9b9a3f2cf49b495e37',
+              tradableInstrument: {
+                __typename: 'TradableInstrument',
+                instrument: {
+                  __typename: 'Instrument',
+                  name: 'Las Vegas nuggets',
+                  code: 'Nuggets2',
+                },
+              },
+            },
+          },
+          {
+            __typename: 'MarketEdge',
+            node: {
+              __typename: 'Market',
+              id: '976f71b8d6f99f3685297156a11245f21e49c99d3be17f2577a4bf693b5f37d1',
+              tradableInstrument: {
+                __typename: 'TradableInstrument',
+                instrument: {
+                  __typename: 'Instrument',
+                  name: 'Nuggets',
+                  code: 'Nuggets',
+                },
+              },
+            },
+          },
+          {
+            __typename: 'MarketEdge',
+            node: {
+              __typename: 'Market',
+              id: '1cb2e1755208914b6f258a28babd19ae8dfbaf4084d8867d8a120c50dca1e17f',
+              tradableInstrument: {
+                __typename: 'TradableInstrument',
+                instrument: {
+                  __typename: 'Instrument',
+                  name: 'CELUSD (June 2022)',
+                  code: 'CELUSD',
+                },
+              },
+            },
+          },
+        ],
+      },
+    },
+    loading: false,
+    error: undefined,
+  })),
+}));
+
+const renderComponent = () =>
+  render(
+    <Router>
+      <MockedProvider mocks={[networkParamsQueryMock]}>
+        <AppStateProvider>
+          <VegaWalletContext.Provider value={mockWalletContext}>
+            <ProposeUpdateMarket />
+          </VegaWalletContext.Provider>
+        </AppStateProvider>
+      </MockedProvider>
+    </Router>
+  );
+
+// Note: form submission is tested in propose-raw.spec.tsx. Reusable form
+// components are tested in their own directory.
+
+describe('Propose Update Market', () => {
+  it('should render successfully', () => {
```

Comment:
suggestion (non blocking): Can we get a test here to actually fill in the fields and submit a tx? Perhaps mock useVegaWallet to detect if the tx is sent correclty. Same for the other form components

Reply:
The test you wrote that's now used in propose-raw.spec.tsx tests submitting a tx. I think we have this covered sufficiently